cmd/snap-update-ns/utils.go

```diff
@@ -286,6 +291,63 @@ func secureMkfileAll(name string, perm os.FileMode, uid, gid int) error {
 	return err
 }
 
+func planWritableMimic(dir string) ([]*Change, error) {
+	// We need a place for "safe keeping" of what is present in the original
+	// directory as we are about to attach a tmpfs there, which will hide
+	// everything inside.
+	logger.Debugf("create-writable-mimic %q", dir)
+	safeKeepingDir := filepath.Join("/tmp/.snap/", dir)
+
+	var changes []*Change
+
+	// Bind mount the original directory elsewhere for safe-keeping.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{
+			// XXX: should we rbind here?
+			Name: dir, Dir: safeKeepingDir, Options: []string{"bind"}},
+	})
+	// Mount tmpfs over the original directory, hiding its contents.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{Name: "none", Dir: dir, Type: "tmpfs"},
+	})
+	// Iterate over the items in the original directory (nothing is mounted _yet_).
+	entries, err := ioutilReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	for _, fi := range entries {
+		ch := &Change{Action: Mount, Entry: mount.Entry{
+			Name: filepath.Join(safeKeepingDir, fi.Name()),
+			Dir:  filepath.Join(dir, fi.Name()),
+			// XXX: should we rbind here?
+			Options: []string{"bind", "ro"},
```

Comment:
Soo... the function is called planWritableMimic, but we are only mounting read-only here. AIUI, the read-only is still hugely useful for making files accessible to where the snap expects them. That said, maybe just call this planMimic()?

Reply:
I was thinking about it and I think the extra "ro" is not needed. Changed.